### ISSUE TITLE
ERROR: Could find any ADB device ERROR: Server connection failed.

### ISSUE THREAD

**Issue by JCSuperFast** [Genymobile/scrcpy#3415]
![image](https://user-images.githubusercontent.com/110431026/182257174-c174bc0b-293c-443d-aa2c-95790c899b5b.png)

Hello im new here 

My phone is Oppo A74
Im trying to open scrcpy in my phone but its not opening showing this error (the photo)
What im gonna do?
(I opened usb debugging in developer settings)

**Comment by rom1v** [comment]
https://github.com/Genymobile/scrcpy/blob/master/FAQ.md#device-not-detected

**Comment by o-data** [comment]
Hi there, I'm having same issue as you JC, still no luck at the moment. I have feeling that Oppo purposely make it difficult or lock it.

I have tried the general Google ADB driver, driver that website advertise as Oppo, various difference version. Turning on and off Developer mode, revoke USB authorize none will find the phone. Various different version of QtScrCpy and SCRCPY, all met with failure.

I also consider trying the WiFi mode, I can see the IP, I know the right IP but SCRCPY still can't connect to it. Maybe I'm doing something wrong.

It work fine on other phone though, but not Oppo one I have. If you do find a solution, please don't forget this post and share.

PS: They also lock the Bootloader too. All online guide out there are generic change but it doesn't work if you follow up.

**Comment by DTubbs** [comment]
Your phone needs to be connected via ADB first. I'm really new to this myself so hopefully anything I get wrong can be corrected by others.

You need  to install Android-platform-tools through your package manager first. This package contains the all of the adb commands you'll need to interact with your android device. Are you USB debugging or wifi debugging?

**Comment by o-data** [comment]
> 
Yes, I'm aware of that DTubbs. Like I said, I can connect with my other phone but not Oppo one, at this point I'm leaning more of an Oppo issue than anything else.

Tried both with USB Debugging and WiFi, none seem to work with ScrCpy.
However, I'm also using MyPhoneExplorer and this one seem to be working but each software have their own different purpose. 
As a trick, since MyPhoneExplorere abd.exe seem to be able to connect perhaps it an adb.exe issue, I copy/paste the file to scrcpy and hoping that it work; well it doesn't, at least not in my test.

Still looking for solution at the moment. I'll give a full installation of the Android-platform-tools once I have some plan, don't want other stuff to break while trying to fix.

**Comment by rom1v** [comment]
You can set the ADB environment variable to the full path (filename included) of another adb.exe if you want to try, scrcpy will use that one instead.

**Comment by ImJING** [comment]
![QQ截图20220813101145](https://user-images.githubusercontent.com/44094039/184464696-9d25a072-3fa9-44e3-93c3-ea086e514da7.png)
adb可以发现我的设备
但是一打开scrcpy就变成这样
win10+sony XZPremium

**Comment by rom1v** [comment]
What is thebresult of `adb devixes -l`?

(please paste the terminal output as text, not a screenshot).

**Comment by MadMatt69** [comment]
i connected the phone with adb but it's not working because he cant find connection i runned the scrpy-console.bat and give me this 
![Immagine 2022-08-20 231048](https://user-images.githubusercontent.com/111661771/185766114-4d778b75-b419-423d-ab5e-7534c44cf866.png)

what i can do?
(im writing here because i cant create a new ticket)

**Comment by rom1v** [comment]
@MadMatt69 `adb devices` indicates that the device is "offline". Check stackoverlow: https://stackoverflow.com/questions/14993855/android-adb-device-offline-cant-issue-commands

**Comment by samheyman** [comment]
Scrcpy was working fine for me a few weeks ago with my Sony Xperia . Today it won't connect:

```PS C:\Users\sam.heyman> scrcpy.exe
scrcpy 1.24 <https://github.com/Genymobile/scrcpy>
ERROR: Could not find any ADB device
ERROR: Server connection failed
PS C:\Users\sam.heyman>```

I haven't changed anything, connected via USB as a developer as per usual.

**Comment by samheyman** [comment]
Yes you are right @rom1v 

```
PS C:\Users\sam.heyman> adb devices
List of devices attached

PS C:\Users\sam.heyman>
```
So adb issue?

**Comment by rom1v** [comment]
@samheyman https://github.com/Genymobile/scrcpy/blob/master/FAQ.md#device-not-detected
Either device, USB cable, drivers, permissions… anything that can cause `adb` not to detect the device.

**Comment by o-data** [comment]
Hi guys, for those that trying to do Wi-FI/Network debug mode, I finally manage to get it to work.

Assuming that you can connect by USB (you have a working Driver), then to do Network you need to use the "Pair" command first then you can use WiFI connection afterward.

Once I have time I will write a more thorough guide

**Comment by samheyman** [comment]
Ok so managed to solve my issue "ERROR: Could not find any ADB device" despite it having been working for months. It was because "Use developer options" on my phone had switched off.
I had to go to Settings > System > Developer options and turn "Use developer options" back on (it was off despite having allowed USB access when connecting the USB and being a developer on the phone).

**Comment by NEU-Apollo** [comment]
> Ok so managed to solve my issue "ERROR: Could not find any ADB device" despite it having been working for months. It was because "Use developer options" on my phone had switched off. I had to go to Settings > System > Developer options and turn "Use developer options" back on (it was off despite having allowed USB access when connecting the USB and being a developer on the phone).

I use OnePlus(一加， OnePlus is a sub-brand of OPPO) and this solution is useful for my phone. I turn "Use developer options" off and then turn it on again then allow USB access. Now my scrcpy can connect to my phone.
